pkg/cmd/init.go

```diff
@@ -278,8 +275,8 @@ func certSetup(cfg *config.MicroshiftConfig) (*certchains.CertificateChains, err
 					"openshift.default",
 					"openshift.default.svc",
 					"openshift.default.svc.cluster.local",
-					"api." + config.DefaultClusterName + "." + cfg.BaseDomain,     // TODO: OpenShift actually uses  api.$ClusterName.$BaseDomain
-					"api-int." + config.DefaultClusterName + "." + cfg.BaseDomain, // TODO: OpenShift actually uses  api.$ClusterName.$BaseDomain
+					"api." + cfg.ClusterName + "." + cfg.BaseDomain,
```

Comment:
According to [OCP documentation](https://docs.openshift.com/container-platform/4.11/installing/installing_platform_agnostic/installing-platform-agnostic.html#installation-dns-user-infra_installing-platform-agnostic), all DNS records are subdomains of `<cluster_name>.<base_domain>`:
```
api.<cluster_name>.<base_domain>
api-int.<cluster_name>.<base_domain>
*.apps.<cluster_name>.<base_domain>
```
And the `openshift-installer`'s [`install-config.yaml`](https://docs.openshift.com/container-platform/4.11/installing/installing_platform_agnostic/installing-platform-agnostic.html#installation-bare-metal-config-yaml_installing-platform-agnostic) takes them in via fields `metadata.name` and `baseDomain`.

In OCP, of course, this is because the cluster-name receives a random suffix to avoid failures when clusters get deleted and re-created under the same name. In MicroShift that won't be the case.

So there isn't really a utility and we could redefine `baseDomain` to also include the cluster name, but my understanding was we wanted to align.

Reply:
I see, so the `spec.baseDomain` field of the config/v1 DNS type is constructed _by the installer_ by concatenating cluster name and the "baseDomain" specified in the install config:

https://github.com/openshift/installer/blob/780b29adbe5ba47a1b41ffeade875583282995db/pkg/asset/manifests/dns.go#L86
https://github.com/openshift/installer/blob/780b29adbe5ba47a1b41ffeade875583282995db/pkg/types/installconfig.go#L196

So, to me, this means that ignoring cluster name entirely would be the choice most consistent with OCP's config/v1 API, perhaps with the sample value for MicroShift's `dns.baseDomain` field being "microshift.example.com".

Reply:
Based on this conversation, I tend to think that we may want to drop the clustername altogether , and just keep the base domain. As the clustername is just a construct of the installer then injected into the cluster itself.

This would leave customers freedom to deploy the way they want.

Reply:
Reverted this.